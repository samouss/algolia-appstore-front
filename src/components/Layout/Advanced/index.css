.Advanced {
  display: flex;
  flex-direction: column;
  margin-top: 1vh;
}

.Advanced__Content {
  margin-bottom: 1vh;
  display: none;

  @media (--tablet) {
    display: flex;
    flex-direction: row;
  }
}

.Advanced__Content--visible {
  display: flex;
  flex-direction: column;
}

.Advanced__Link {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 25px;
  color: #b9b9b9;
  font-size: 0.8rem;

  @media (--tablet) {
    display: none;
  }
}
