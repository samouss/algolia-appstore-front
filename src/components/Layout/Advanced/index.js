import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './index.css';

class Advanced extends Component {

  constructor(props) {
    super(props);

    this.state = {
      isVisible: false,
    };

    this.toggle = this.toggle.bind(this);
  }

  toggle(event) {
    event.preventDefault();

    this.setState(state => ({
      isVisible: !state.isVisible,
    }));
  }

  render() {
    const { isVisible } = this.state;
    const { children } = this.props;
    const icon = isVisible ? <i className="fa fa-chevron-up" aria-hidden="true" />
      : <i className="fa fa-chevron-down" aria-hidden="true" />;

    return (
      <div styleName="Advanced">
        <div
          styleName={cx('Advanced__Content', {
            'Advanced__Content--visible': isVisible,
          })}
        >
          {children}
        </div>
        <a
          href=""
          styleName="Advanced__Link"
          onClick={this.toggle}
        >
          {!isVisible && 'Advanced filters'}
          {icon}
        </a>
      </div>
    );
  }

}

Advanced.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Advanced;
