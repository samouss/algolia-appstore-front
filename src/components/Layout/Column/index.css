.Column {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  flex-wrap: wrap;
  width: 100%;
  margin-top: 1vh;
  // background-color: blue;
  // border: 1px solid #f2f2f2;

  &:first-child {
    margin-top: 0;
    margin-left: 0;
  }

  @media (--tablet) {
    width: 25%;
    margin-top: 0;
    margin-left: 1vh;
  }
}
