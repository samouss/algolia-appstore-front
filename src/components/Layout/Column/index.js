import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const Column = ({ children }) => (
  <div styleName="Column">
    {children}
  </div>
);

Column.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Column;
