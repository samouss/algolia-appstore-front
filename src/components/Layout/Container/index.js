import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const Container = ({ children }) => (
  <div styleName="Container">
    {children}
  </div>
);

Container.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Container;
