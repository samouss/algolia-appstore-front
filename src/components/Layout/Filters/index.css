.Filters {
  margin-bottom: 2.5vh;
}
