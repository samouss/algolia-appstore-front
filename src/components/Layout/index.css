.Layout {
  display: flex;
  flex-direction: column;
  width: 90vw;
  max-width: 768px;
  min-height: 95vh;
  margin: 2.5vh auto;
}
