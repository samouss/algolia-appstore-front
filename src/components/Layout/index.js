export { default as Advanced } from './Advanced';
export { default as Row } from './Row';
export { default as Column } from './Column';
export { default as Container } from './Container';
export { default as Filters } from './Filters';
