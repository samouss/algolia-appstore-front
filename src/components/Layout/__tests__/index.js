import React from 'react';
import { shallow } from 'enzyme';
import Layout from '../index';

describe('<Layout />', () => {
  const defaultProps = {};

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <Layout
        {...props}
      >
        Content
      </Layout>,
    );

    expect(component).toMatchSnapshot();
  });
});
