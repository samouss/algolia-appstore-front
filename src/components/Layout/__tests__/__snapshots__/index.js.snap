// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> expect to render 1`] = `
<div
  styleName="Container"
>
  Content
</div>
`;

exports[`<Layout /> expect to render 1`] = `
<div
  styleName="Layout"
>
  Content
</div>
`;
